### PR TITLE
Sdk final layer: use `builder` user

### PR DESCRIPTION
**Pull request by jpmcb**
**Issue number:**

N/a - related to permissions errors in final artifacts when building with SDK

Related to changes from #100 

**Description of changes:**

```
- Final sdk layer: use `builder` user
- Upgrade to v0.30.2
```

**Testing done:**

Sanity check: Built locally, launched a container and see the correct user
```
❯ docker run --rm -it bottlerocket/sdk-x86_64:v0.30.1-x86_64
[builder@e16284e24d24 ~]$ whoami
builder
[builder@e16284e24d24 ~]$ pwd
/home/builder
```

**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1069,3 +1069,10 @@ CMD ["/bin/bash"]
 FROM scratch as sdk-golden
 
 COPY --from=sdk-final / /
+
+# The `builder` user is setup in the "final" layer
+# and is used in place of the default `root` user
```
Comment (bcressey) [inline Dockerfile]: nit: the line length here seems kind of parsimonious

File: Dockerfile [hunk 2]
```diff
@@ -1069,3 +1069,10 @@ CMD ["/bin/bash"]
...
+USER builder
+WORKDIR /home/builder
+
+CMD ["/bin/bash"]
```
Comment (bcressey) [inline Dockerfile:1076]: nit: can omit the earlier `CMD` line

**Review by stmcginnis — APPROVED** [review]
Looks like all comments have been addressed. Changes look good to me.

**Review by etungsten — APPROVED** [review]

**Review by jpculp — APPROVED** [review]
- [x] Build SDK (and Bottlerocket) `x86_64` on `x86_64`
- [x] Build SDK (and Bottlerocket) `aarch64` on `x86_64`
- [x] Build SDK (and Bottlerocket) `aarch64` on `aarch64`
- [x] Build SDK (and Bottlerocket) `x86_64` on `aarch64`

Basic smoke tests on `aws-k8s-1.24` (x86_64) also passed.

**Pull request merged**